Simplify build for different auth strategies

vidakovic opened the pull request:
## Description

The current process of selecting the authentication strategy for Fineract is unnecessarily complicated. Spring profiles are not the best way to enable/disable these features. By using "@ConditionalOnProperty" we only need one build and can enable the required authentication scheme with environment variables. This makes maintenance of the application.properties file also easier (just one instead of 4 different variants), and in turn makes the build script simpler.

vidakovic committed:
Simplify build for different auth strategies

vidakovic commented:
@ptuomola just FYI: please approve and merge #2009 first

ptuomola commented on .github/workflows/build.yml:
```diff
@@ -47,5 +56,24 @@ jobs:
+      - name: Basic Auth Build & Test
+        env:
+          - FINERACT_SECURITY_BASICAUTH_ENABLED: true
+          - FINERACT_SECURITY_OAUTH_ENABLED: false
```
Do these variables actually do anything? It seems that we are hardcoding the same parameters in the command line when starting the server for oauth2-test and basictest etc? So it would seem that these are redundant... correct?

vidakovic replied:
Hm, now that you mention it... actually no... I was starting with a different idea and then I guess I forgot to remove them. Good catch!

ptuomola commented on oauth2-tests/build.gradle:
```diff
@@ -45,7 +45,7 @@ cargo {
         startStopTimeout = 240000
         containerProperties {
-            property 'cargo.start.jvmargs', '-Dspring.profiles.active=oauth'
+            property 'cargo.start.jvmargs', '-Dfineract.security.basicauth.enabled=false -Dfineract.security.oauth.enabled=true -Dfineract.security.2fa.enabled=false'
```
Ie the environment is hardcoded here and these are the values used - correct?

vidakovic replied:
Correct, exactly the same behavior as before, just with properties instead of profiles.

ptuomola reviewed (changes requested):
Two further questions/comments:
- I think the README and oauth etc documentation should also be updated to reflect this new set-up? 
- Some of the settings are mutually exclusive.. i.e. you can't have both basicauth and oauth enabled at the same time. Should we detect this and fail the start-up?

vidakovic commented:
I'll update the README. Good idea with the startup validation, I like that idea.

vidakovic commented:
@ptuomola ... alright, added documentation and configuration checks... and did the requested cleanups.

vidakovic commented:
@ptuomola Hi Petri... finally... I think this should run through now properly... added a couple of last minute fixes to make the unit tests run again properly, do some minor cleanup (e. g. `TwoFactorUtil` not used anymore). I've also rebased with the latest in `develop`. As soon as this one is in I'll start the migration to Testcontainers. Would be great if you could approve and merge.

ptuomola reviewed (approved):
LGTM - great job!
